src/arrayiterator.vala

```diff
@@ -46,12 +46,26 @@ namespace Vast {
             return !this.ended;
         }
 
-        public unowned T get() {
-            return (T) ((uint8*) array.data.get_data () + offset);
+        public void* get()
```

Comment:
probably shall change these methods to match the name in Array. 

-- `get` is triggered by vala foreach, but we will likely never really use the feature; we will do multi array iteration for the universal functions.